src/Commands/Plumbing/Commit/CommitFromTree.hs

```diff
@@ -8,10 +8,10 @@ import Data.Time
 import qualified Data.ByteString.Char8 as B
 
 cmdCommitFromTree :: Command
-cmdCommitFromTree (author:email:tree:parents) = do
+cmdCommitFromTree (au:em:tree:pars) = do
   now <- getZonedTime
   msg <- getContents
-  let commit = Commit (B.pack tree) (map B.pack parents) author email now msg
+  let commit = Commit (B.pack tree) (map B.pack pars) au em now msg
```

Comment:
I'd be hard pressed to say they are good practice anywhere, to be quite frank. While nigh ubiquitous (pick any random package from Hackage and upon inspecting its source code you're likely to find functions that have arguments with single letter names), I'd much rather have a descriptive name, even if a little long.

In this particular case `parents`, `author` and `email` sadly shadow the bindings created by using the record syntax, and thus generate warnings. For clarity when reading the code, it might be best not to name-shadow, so maybe take a page out of the Lens book and update the type definition somewhat like this?

```haskell
data Commit = Commit { _author :: String
                     , _email :: String
                     -- ...
                     }
```

What do you think?